magic.js

```diff
@@ -724,19 +707,23 @@ function sign(message, sk, cb) {
       sk  = sodium.crypto_sign_ed25519_sk_to_seed(isk);
   }
 
-  let signature;
-  try {
-    signature = sodium.crypto_sign_detached(payload, isk);
-  } catch(ex) {
-    return done(new Error('Libsodium error: ' + ex));
-  }
-
-  return done(null, convert({
-    alg:       'ed25519',
-    sk:        sk,
-    payload:   payload,
-    signature: signature
-  }));
+  const key = {
+    key: Buffer.concat([
+      Buffer.from('302e020100300506032b657004220420', 'hex'),
+      sk.subarray(0, 32),
```

Comment:
Because only the first 32 bytes is the value of the private key, the rest is "libsodium" format. What happens here is I take the 32 bytes that are the private key and prefix that with a PKCS8 ASN.1 structure that has the id-Ed25519 OID so that it can be used by the crypto module.

I've kept the lines 698-707 so that the key is can still be passed as either secret key bytes, seedbytes, or generated new. I also didn't want to change the function return which contains `sk`.